pkg/chains/storage/docdb/docdb.go

```diff
@@ -179,11 +172,21 @@ func WatchBackend(ctx context.Context, cfg config.Config, watcherStop chan bool)
 		}
 	}()
 
-	// Add a path.
-	err = watcher.Add(cfg.Storage.DocDB.MongoServerURLDir)
-	if err != nil {
-		return nil, err
+	if cfg.Storage.DocDB.MongoServerURLPath != "" {
+		dirPath := filepath.Dir(cfg.Storage.DocDB.MongoServerURLPath)
+		// Add a path.
+		err = watcher.Add(dirPath)
+		if err != nil {
+			return nil, err
+		}
+	} else if cfg.Storage.DocDB.MongoServerURLDir != "" {
```

Comment:
I'm wondering if this should just be an `if-block` instead of a `else-if-block`. I'm thinking of the case where the config changes. It's probably safer to watch all the paths that are not an empty string.

Reply:
updated!